shpc/main/modules/templates/docker.lua

```diff
@@ -54,7 +54,7 @@ local inspectCmd = "{{ command }} ${PODMAN_OPTS} inspect ${PODMAN_COMMAND_OPTS}
 set_shell_function("{|module_name|}-shell", shellCmd,  shellCmd)
 
 -- conflict with modules with the same name
-conflict(myModuleName(){% if aliases %}{% for alias in aliases %}{% if alias.name != name %},"{{ alias.name }}"{% endif %}{% endfor %}{% endif %})
+conflict({% if name != tool %}"{{ name }}",{% endif %}"{{ tool }}"{% if aliases %}{% for alias in aliases %}{% if alias.name != tool %},"{{ alias.name }}"{% endif %}{% endfor %}{% endif %})
```

Comment:
I don't see how this is a better fix? Where does tool come from (note it could be undefined if it's set here https://github.com/singularityhub/singularity-hpc/blob/efb3ff13c653455525a54e2cc0e90111e567c34b/shpc/main/modules/__init__.py#L44) and wouldn't it be an issue if the module name is mypython and that's provided by another module (so we would just want to conflict the module name regardless as we do now). And also the if logic is changed so there will be a hanging comma if no aliases.

Reply:
I think that here tool is unrelated to the naming convention that can be set in the settings file.

It seems to me that `tool` comes from the template rendering call:   https://github.com/singularityhub/singularity-hpc/blob/main/shpc/main/container/singularity.py#L185

and hence from `parsed_name.tool` . Here, `parsed_name` seems to originally come out of the attributes of the ContainerName class:   https://github.com/singularityhub/singularity-hpc/blob/main/shpc/main/container/base.py#L15-L38

Putting more comments below..